app/src/behaviors/behavior_sticky_key.c

```diff
@@ -0,0 +1,272 @@
+    int timer_cancel_result = k_delayed_work_cancel(&sticky_key->release_timer);
+    if (timer_cancel_result == -EINPROGRESS) {
+        // too late to cancel, we'll let the timer handler clear up.
+        sticky_key->timer_is_cancelled = true;
+    }
+    return timer_cancel_result;
+}
+
+static int on_sticky_key_binding_pressed(struct zmk_behavior_binding *binding,
+                                         struct zmk_behavior_binding_event event) {
+    struct device *dev = device_get_binding(binding->behavior_dev);
+    const struct behavior_sticky_key_config *cfg = dev->config_info;
+
+    struct active_sticky_key *sticky_key =
+        store_sticky_key(event.position, binding->param1, binding->param2, cfg);
+    if (sticky_key == NULL) {
+        LOG_ERR("unable to store sticky key, did you press more than %d sticky_key?",
+                ZMK_BHV_STICKY_KEY_MAX_HELD);
+        return 0;
+    }
+
+    press_sticky_key_behavior(sticky_key, event.timestamp);
+    LOG_DBG("%d new sticky_key", event.position);
+    return 0;
+}
+
+static int on_sticky_key_binding_released(struct zmk_behavior_binding *binding,
+                                          struct zmk_behavior_binding_event event) {
+    struct active_sticky_key *sticky_key = find_sticky_key(event.position);
+    if (sticky_key == NULL) {
+        LOG_ERR("ACTIVE STICKY KEY CLEARED TOO EARLY");
+        return 0;
+    }
+
+    if (sticky_key->modified_key_usage_page != 0 && sticky_key->modified_key_keycode != 0) {
+        LOG_DBG("Another key was pressed while the sticky key was pressed. Act like a normal key.");
+        int retval = release_sticky_key_behavior(sticky_key, event.timestamp);
+        clear_sticky_key(sticky_key);
+        return retval;
+    }
+
+    // No other key was pressed. Start the timer.
+    sticky_key->timer_started = true;
+    sticky_key->release_at = event.timestamp + sticky_key->config->release_after_ms;
+    // adjust timer in case this behavior was queued by a hold-tap
+    s32_t ms_left = sticky_key->release_at - k_uptime_get();
+    if (ms_left > 0) {
+        k_delayed_work_submit(&sticky_key->release_timer, K_MSEC(ms_left));
+    }
+    return 0;
+}
+
+static const struct behavior_driver_api behavior_sticky_key_driver_api = {
+    .binding_pressed = on_sticky_key_binding_pressed,
+    .binding_released = on_sticky_key_binding_released,
+};
+
+static int sticky_key_keycode_state_changed_listener(const struct zmk_event_header *eh) {
+    if (!is_keycode_state_changed(eh)) {
+        return 0;
+    }
+    struct keycode_state_changed *ev = cast_keycode_state_changed(eh);
+    for (int i = 0; i < ZMK_BHV_STICKY_KEY_MAX_HELD; i++) {
+        struct active_sticky_key *sticky_key = &active_sticky_keys[i];
+        if (sticky_key->position == ZMK_BHV_STICKY_KEY_POSITION_NOT_USED) {
+            continue;
+        }
+
+        if (strcmp(sticky_key->config->behavior.behavior_dev, "KEY_PRESS") == 0 &&
```

Comment:
Hmm... I don't love keying off this label, but this might be the pragmatic way to do this. Any other options you'd considered?

Reply:
Previously I had this keyed on key_position, but that requires `keycode_state_changed` to gain a `position` field too.

Reply:
I think that's fine, because the `zmk_behavior_binding_event` already has the position, so it can just be passed on and used as a unique identifier.

Reply:
Let's leave it then, and consider circling back on this later.